sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1460,6 +1469,141 @@ public Mono<Void> rollbackTransaction(ServiceBusTransactionContext transactionCo
+     * The actual number of deleted messages may be less if there are fewer eligible messages in the entity.
+     * <p>If the lock for a message is held by a receiver, it will be respected and the message will not be deleted.</p>
+     * <p>You can delete a maximum of 4000 messages in a single API call, this is the current limit determined by
+     * the Service Bus service.</p>
+     *
+     * @param messageCount the desired number of messages to delete.
+     * @return a {@link Mono} indicating the number of messages deleted.
+     *
+     * @throws IllegalArgumentException when the {@code messageCount} is less than 1 or exceeds the maximum allowed, as
+     * determined by the Service Bus service.
+     */
+    public Mono<Integer> deleteMessages(int messageCount) {
+        return deleteMessages(messageCount, new DeleteMessagesOptions(), receiverOptions.getSessionId());
+    }
+
+    /**
+     * Deletes up to {@code messageCount} messages from the entity. The actual number of deleted messages may be less
+     * if there are fewer eligible messages in the entity.
+     * <p>If the lock for a message is held by a receiver, it will be respected and the message will not be deleted.</p>
+     * <p>You can delete a maximum of 4000 messages in a single API call, this is the current limit determined by
+     * the Service Bus service.</p>
+     *
+     * @param messageCount the desired number of messages to delete.
+     * @param options options used to delete the messages.
+     * @return a {@link Mono} indicating the number of messages deleted.
+     *
+     * @throws IllegalArgumentException when the {@code messageCount} is less than 1 or exceeds the maximum allowed, as
+     * determined by the Service Bus service.
+     * @throws NullPointerException if {@code options} is null.
+     */
+    public Mono<Integer> deleteMessages(int messageCount, DeleteMessagesOptions options) {
+        return deleteMessages(messageCount, options, receiverOptions.getSessionId());
+    }
+
+    /**
+     * Attempts to purge all messages from an entity.
+     * <p>If the lock for a message is held by a receiver, it will be respected and the message will not be deleted.</p>
+     * <p>
+     * This method may invoke multiple service requests to delete all messages. Because multiple service requests may be
+     * made, the possibility of partial success exists, in such scenario, the method will stop attempting to delete
+     * additional messages and throw the exception that was encountered. Also, due to the multiple service requests,
+     * purge operation may exceed the configured {@link AmqpRetryOptions#getTryTimeout()}.
+     * </p>
+     *
+     * @return a {@link Mono} indicating the number of messages deleted.
+     */
+    public Mono<Integer> purgeMessages() {
+        return purgeMessages(new PurgeMessagesOptions());
+    }
+
+    /**
+     * Attempts to purge all messages from an entity.
```

Comment:
Would it make sense to call out that it deletes all messages that were created before configured timestamp or `now`? Otherwise it might sound like a scary endless loop :)

Reply:
Yes, good point!, will update the java-doc

Reply:
done